src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html

```diff
@@ -534,25 +534,43 @@ <h4 *ngIf="isImport" id="jhi-programming-exercise-heading-import" jhiTranslate="
                 </label>
             </div>
         </div>
-        <div class="pt-4"></div>
-        <div class="form-group" *ngIf="programmingExercise.id && !isImport">
-            <label class="form-control-label" jhiTranslate="artemisApp.exercise.notificationText" for="field_notification_text">Notification Text</label>
-            <input minlength="3" type="text" class="form-control" name="notificationText" id="field_notification_text" [(ngModel)]="notificationText" />
+        <div class="update-programming-exercise-footer">
+            <div class="update-programming-exercise-footer-content">
+                <span *ngIf="isSaving" jhiTranslate="artemisApp.quizExercise.edit.saving" class="badge bg-secondary"> </span>
+                <ng-template #tooltipTranslate>
+                    <div *ngFor="let reason of getInvalidReasons()">
+                        <p
+                            *ngIf="(reason.translateValues | json) != '{}'"
```

Comment:
This could probably be simplyfied to `Object.entries(reason.translateValues).length === 0;`

Or could it be possible to just not have this check at all? I think you should be fine by passing an empty object to [translateValues]

Reply:
You are right, we do not need the translateValues as they are not used here.